### PR TITLE
Warn about long delays

### PR DESCRIPTION
**Pull request by paulmenzel**

### CODE REVIEW

File: source/components/executer/exsystem.c [hunk 1]
```diff
@@ -338,6 +338,12 @@ AcpiExSystemDoSleep (
 
     AcpiExExitInterpreter ();
 
+    if (HowLongMs > 50)
```
Comment (rafaeljw) [inline source/components/executer/exsystem.c:341]: Shouldn't this be HowLongMs > 10?

**Review by rafaeljw — CHANGES_REQUESTED** [review]
If this is a spec violation, the commit log should at least mention which section of the spec is violated and how.

File: source/components/executer/exsystem.c (hunk 1 above)
Comment (paulmenzel) [reply source/components/executer/exsystem.c]: Indeed. Fixed.

**Comment by paulmenzel** [comment]
> If this is a spec violation, the commit log should at least mention which section of the spec is violated and how.

Only the stall is a specification violation.

The sleep warning is just for good implementation, so sleeps longer than 10 ms hopefully get reimplemented by a polling approach or something similar.

**Comment by acpibob** [comment]
I’m seeing some long sleeps in existing ASL code, some examples:

MSI/MS/MS-7752/14FAD414B696:                Sleep (0x64)
TONK/TN/TN1402/ABC6298CB633:                Sleep (0x64)
ZOTAC/NM/NM10/27F12A855946:            Sleep (0x0BB8)
TONK/C/C31/7F7E8D75C7FF:                                    Sleep (0xFA)
Supermicro/X8/X8SIL/40AECBFF4573:                    Sleep (0x0BB8)
MSI/MS-7/MS-7A38/53B5D217D45C:            Sleep (0x012C)
MSI/MS/MS-7721/3EF15301B047:                                Sleep (0xFA)

**Comment by paulmenzel** [comment]
[If replying to GitHub emails, please always remove the citation.]

> I’m seeing some long sleeps in existing ASL code, some examples:

>     MSI/MS/MS-7752/14FAD414B696:                Sleep (0x64)
>     TONK/TN/TN1402/ABC6298CB633:                Sleep (0x64)
>     ZOTAC/NM/NM10/27F12A855946:            Sleep (0x0BB8)
>     TONK/C/C31/7F7E8D75C7FF:                                    Sleep (0xFA)
>     Supermicro/X8/X8SIL/40AECBFF4573:                    Sleep (0x0BB8)
>     MSI/MS-7/MS-7A38/53B5D217D45C:            Sleep (0x012C)
>     MSI/MS/MS-7721/3EF15301B047:                                Sleep (0xFA)

Yes, and these should be warned/notified about. For example, 0x64 ms is 100 ms, 0xfa ms is 250 ms and 0xbb8 ms is 3000 ms greater than the current 2 second maximum. It’s non-optimal, that means bad, implementation in firmware, and gives a bad experience. (The question remains, if the OS like GNU/Linux hits these code paths.)

**Comment by paulmenzel** [comment]
> This of course means that these machines will always generate warnings from ACPI.

Did you test, that these code paths from your examples are actually executed (all the time). Nevertheless, if these delays are always happening, the user should be made aware of them and the firmware should be fixed, so it’s exactly the wanted outcome of these changes.

> This is discouraged, especially in Linux.

Where is that documented, that it’s discouraged? Elaborate log messages, and actionable warnings are encouraged in my experience.

> Perhaps implementing this warning in the iASL compiler would be better.

It would not hurt, but then only folks using an iASL with that patch will see the warning, and would be able to fix it. The user should still be informed about the bad implementation.

**Comment by acpibob** [comment]
No, I didn’t test for actual execution.

We can try adding this warning, but if we get too many complaints about it, we will need to back it out.

**Comment by paulmenzel** [comment]
Understood.

By the way, https://bugzilla.kernel.org/show_bug.cgi?id=208705 motivated me to look into this. Also to figure out, which of these sleeps is actually executed.

**Comment by acpibob** [comment]
Yes, this is ok. Probably split it into 2 lines, however.

**Comment by paulmenzel** [comment]
I force pushed the improved commits.

**Review by rafaeljw — APPROVED** [review]
All 3 commits look good to me now.

**Comment by acpibob** [comment]
Ok, got it. Thanks, Bob

**Comment by paulmenzel** [comment]
Sorry for the mistakes that crept in, and thank you all for fixing them:

1.  commit https://github.com/acpica/acpica/commit/441747f1dcff770d692acbfd4d85b2cfaabdb38a
2.  commit https://github.com/acpica/acpica/commit/82a46ba57fe03ae99342740b92a04d8a8184860d

For the record, on the Acer TravelMate 5735Z/BA51_MV, BIOS V1.14 07/26/2011, there are some sleeps related to brightness configuration, that take 100 ms (0x64):

```
[   11.169493] IPv6: ADDRCONF(NETDEV_CHANGE): enp5s0: link becomes ready
[   11.984097] ACPI Warning: Firmware issue: Excessive sleep time (0x0000000000000064 ms > 10 ms) in ACPI Control Method (20220331/exsystem-177)
[   12.092233] ACPI Warning: Firmware issue: Excessive sleep time (0x0000000000000064 ms > 10 ms) in ACPI Control Method (20220331/exsystem-177)
[   12.200228] ACPI Warning: Firmware issue: Excessive sleep time (0x0000000000000064 ms > 10 ms) in ACPI Control Method (20220331/exsystem-177)
[   12.309446] ACPI Warning: Firmware issue: Excessive sleep time (0x0000000000000064 ms > 10 ms) in ACPI Control Method (20220331/exsystem-177)
[   12.417404] ACPI Warning: Firmware issue: Excessive sleep time (0x0000000000000064 ms > 10 ms) in ACPI Control Method (20220331/exsystem-177)
[   12.526249] ACPI Warning: Firmware issue: Excessive sleep time (0x0000000000000064 ms > 10 ms) in ACPI Control Method (20220331/exsystem-177)
[   13.566840] rfkill: input handler disabled
```

**Pull request merged**